lib/Sema/CSSimplify.cpp

```diff
@@ -6694,6 +6697,22 @@ static ConstraintFix *maybeWarnAboutExtraneousCast(
   // "from" could be less optional than "to" e.g. `0 as Any?`, so
   // we need to store the difference as a signed integer.
   int extraOptionals = fromOptionals.size() - toOptionals.size();
+
+  // From expression could be a type variable with a value to optional
```

Comment:
Super-nit: Write From into quotes or back-ticks here, got me confused initially :) `a` is extraneous since restrictions is plural.